cargo-dist/src/lib.rs

```diff
@@ -57,8 +57,90 @@ pub mod tasks;
 #[cfg(test)]
 mod tests;
 
+/// dist env test -- make sure we have everything we need for a build.
+pub fn do_env_test(cfg: &Config) -> DistResult<()> {
+    let (dist, _manifest) = tasks::gather_work(cfg)?;
+
+    let builds = dist.config.builds;
+
+    let need_cargo_auditable = builds.cargo.cargo_auditable;
+    let need_cargo_cyclonedx = builds.cargo.cargo_cyclonedx;
+    let need_omnibor = builds.omnibor;
+    let mut need_xwin = false;
+    let mut need_zigbuild = false;
+
+    let tools = dist.tools;
+
+    let cargo = tools.cargo()?;
+    let host = cargo.host_target.parse()?;
+
+    for step in dist
+        .local_build_steps
+        .iter()
+        .chain(dist.global_build_steps.iter())
+    {
+        // Can't require cross-compilation tools if we aren't compiling.
+        if cfg.artifact_mode == ArtifactMode::Lies {
+            break;
+        }
+
+        match step {
+            BuildStep::Cargo(step) => {
+                let target = step.target_triple.parse()?;
+                let wrapper = tasks::build_wrapper_for_cross(&host, &target)?;
+
+                match wrapper {
+                    Some(CargoBuildWrapper::Xwin) => {
+                        need_xwin = true;
+                    }
+                    Some(CargoBuildWrapper::ZigBuild) => {
+                        need_zigbuild = true;
+                    }
+                    None => {}
+                }
+            }
+            _ => {}
+        }
+    }
+
+    // These are all of the tools we can check for.
+    //
+    // bool::then(f) returns an Option, so we start with a
+    // Vec<Option<Result<&Tool, DistResult>>>.
+    let all_tools: Vec<Option<DistResult<&Tool>>> = vec![
+        need_cargo_auditable.then(|| tools.cargo_auditable()),
```

Comment:
We should try to scope these to whether we're building the phase(s) that requires them. eg if we're only doing global builds, we shouldn't fail on `auditable` being missing.

Reply:
I made `needs_cargo_auditable` and `needs_cargo_cyclonedx` only true if building local artifacts, which should address this.